modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java

```diff
@@ -122,25 +124,23 @@ public SELF withInitScript(String initScriptPath) {
         return self();
     }
 
+    @SneakyThrows(InterruptedException.class)
     @Override
     protected void waitUntilContainerStarted() {
         logger().info("Waiting for database connection to become available at {} using query '{}'", getJdbcUrl(), getTestQueryString());
 
         // Repeatedly try and open a connection to the DB and execute a test query
         long start = System.currentTimeMillis();
-        try {
-            while (System.currentTimeMillis() < start + (1000 * startupTimeoutSeconds)) {
-                try {
-                    if (!isRunning()) {
-                        Thread.sleep(100L);
-                        continue; // Don't attempt to connect yet
-                    }
 
-                    try (Connection connection = createConnection("")) {
-                        boolean testQuerySucceeded = connection.createStatement().execute(this.getTestQueryString());
-                        if (testQuerySucceeded) {
-                            break;
-                        }
+        while (System.currentTimeMillis() < start + (1000 * startupTimeoutSeconds)) {
```

Comment:
Just an idea: 
I wonder if `Unreliables` can be used in this case to reduce complexity of all that `while sleep` construction ? @bsideup @rnorth @kiview whats your opinion on that ? 

I know that it was historically written that way, but imho that would really helped if that test would be rewritten without any sleep in here

Reply:
This is a good question, maybe @rnorth has an idea, why `Unrealiables` isn't a good idea here. Else, I would suppose it is a good change.

Reply:
I second that!

Reply:
I don't believe that Unreliables would work here since there is the clause:
`catch (NoDriverFoundException e) {
                    // we explicitly want this exception to fail fast without retries
                    throw e;
                }`

and Unreliables waits until a timeout is reached no matter what. Don't have time now to check, but probably using Unreliables might make it only more complicated.